### PR TITLE
Feature/fetch template form

### PR DESCRIPTION
**Pull request by danielniebles**
Hi, please you help reviewing these changes

### CODE REVIEW

File: src/forms/forms.service.ts [hunk 1]
```diff
@@ -14,4 +15,36 @@ export class FormsService {
...
+      return new Promise((resolve) => 
+      setTimeout( () => {
+        resolve(ListsDB[item])
+      }, 0))
+    }
+
+    const listsArray = await Promise.all( lists.map( async (list) => {
+      const response = await fetchDatabase(list)
+      return response
+    }))
```
Comment (CamilaNiebles) [inline src/forms/forms.service.ts:32]: `const listArray = await Promise.all( lists.map ( (list) => fetchDatabase(list)))`

File: src/forms/forms.service.ts [hunk 2]
```diff
@@ -1,12 +1,36 @@
...
+
+  async getTemplateByName(name: string){
+    const response = await this.templateRepository.getTemplateByName(name)
+    const { question } = response
+    const includesList = question.filter( (question) => question['type'].includes('list') )
+    const lists = [...new Set(includesList.map((question) => question['resource']))]
+
+    const listsArray = await Promise.all( lists.map( (list) => {
+      return this.listRepository.getListByName(list)
+    }))
```
Comment (CamilaNiebles) [inline src/forms/forms.service.ts]: You are working in a getTemplateByName scope so, this logic is another responsibility maybe, you should move this logic in an isolated piece of code to improve your readability and the unit test.

File: src/forms/forms.service.ts [hunk 3]
```diff
@@ -1,12 +1,36 @@
...
+
+    const finalForm = question.map( (question) => {
+      const listItems = listsArray.filter( (list) => list['name'] === question['resource'])
+      question['options'] = listItems.length > 0 ? listItems[0]['options'] : null
+      return question
+      }
+    )
```
Comment (CamilaNiebles) [inline src/forms/forms.service.ts]: The same with that.

File: src/forms/forms.service.ts (hunk 2 above)
Comment (danielniebles) [reply src/forms/forms.service.ts]: Moved to manageListsContent function

File: src/forms/forms.service.ts (hunk 3 above)
Comment (danielniebles) [reply src/forms/forms.service.ts]: Moved to generateFinalForm function

**Pull request merged**